sycl/include/sycl/detail/accessor_impl.hpp

```diff
@@ -128,19 +131,24 @@ class __SYCL_EXPORT AccessorImplHost {
   // Outdated, leaving to preserve ABI.
   // TODO: Remove during next major release.
   bool MIsESIMDAcc;
+
+  // To preserve runtime properties
+  property_list MPropertyList;
 };
 
 using AccessorImplPtr = std::shared_ptr<AccessorImplHost>;
 
 class AccessorBaseHost {
 public:
+  template <typename PropertyListT = property_list>
   AccessorBaseHost(id<3> Offset, range<3> AccessRange, range<3> MemoryRange,
                    access::mode AccessMode, detail::SYCLMemObjI *SYCLMemObject,
                    int Dims, int ElemSize, int OffsetInBytes = 0,
-                   bool IsSubBuffer = false) {
+                   bool IsSubBuffer = false,
+                   const PropertyListT &PropertyList = {}) {
     impl = std::shared_ptr<AccessorImplHost>(new AccessorImplHost(
         Offset, AccessRange, MemoryRange, AccessMode, SYCLMemObject, Dims,
-        ElemSize, OffsetInBytes, IsSubBuffer));
+        ElemSize, OffsetInBytes, IsSubBuffer, false, PropertyList));
```

Comment:
Why the second to last argument is `false` here, what does it mean? Should it be `IsSubBuffer`?

Reply:
https://github.com/intel/llvm/pull/6614/files#diff-ee8a1a8bbf896cc0c877b1d0e9ca1a13963f5c80b979f49c20a50b540399c4f9R83 it's to define IsESIMDAcc which is false by default.

Reply:
Thanks, it could be useful to add a comment like `/* IsESIMDAcc = */`, but I will leave that up to you (or better someone from RT folks to decide whether it is needed or not

Reply:
Since you have a question, I believe it will not be superfluous, thanks [6d19248](https://github.com/intel/llvm/pull/6614/commits/6d1924846ce72d04b390a4fcc3ecfcd028274eef)